### PR TITLE
[GPU] Add blocked layouts support to bucketize

### PR DESCRIPTION
**Pull request by tgubanova-lohika**
- changed bucketize kernel
- changed unit tests to check layouts

### Tickets:
 - 87568

### CODE REVIEW

**Comment by mhnap-lohika** [comment]
Maybe add [GPU] prefix to the PR name?

File: src/plugins/intel_gpu/tests/test_cases/bucketize_gpu_test.cpp [hunk 1]
```diff
@@ -41,25 +46,34 @@ struct bucketize_test : testing::TestWithParam<bucketize_test_params<I, B, O>> {
...
+        topology.add(
+            reorder("plane_bucketize_left_bound", "bucketize_left_bound", format::bfyx, type_to_data_type<O>::value));
 
         network network(engine, topology);
         network.set_input_data("input", input);
         network.set_input_data("buckets", buckets);
         const auto outputs = network.execute();
 
-        EXPECT_EQ(outputs.size(), size_t(2));
+        //   EXPECT_EQ(outputs.size(), size_t(2));
```
Comment (mhnap-lohika) [inline src/plugins/intel_gpu/tests/test_cases/bucketize_gpu_test.cpp:73]: Maybe delete or uncomment this line?

Comment (tgubanova-lohika) [reply src/plugins/intel_gpu/tests/test_cases/bucketize_gpu_test.cpp:73]: will remove, as it doesn't make big sense

**Comment by wilson-seok** [comment]
No performance regression during int8 models.

**Comment by wilson-seok** [comment]
@tgubanova-lohika Could you please rebase the PR? I cannot trigger rebuild of gpu_windows_vs2019_release because the page is expired. Sorry for late review.

**Review by wilson-seok — APPROVED** [review]

**Review by yeonbok — APPROVED** [review]

**Pull request merged**